### PR TITLE
chore(build): use `tsx` to run build scripts

### PR DESCRIPTION
**Pull request by jtoar**
@Tobbe refactored all the build scripts to use tsx

### CODE REVIEW

File: packages/cli-helpers/build.ts
Comment (jtoar) [inline packages/cli-helpers/build.ts:1]: For some reason git thinks that I moved this file from `./packages/project-config`. 🤷 I didn't; all I did was rename it from the file right beside it (`./build.js`) to `.ts`

File: packages/project-config/build.ts
Comment (jtoar) [inline packages/project-config/build.ts:1]: See https://github.com/redwoodjs/redwood/pull/9941/files#r1472076586

File: packages/project-config/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "dist"
   ],
   "scripts": {
-    "build": "yarn node ./build.js && run build:types",
```
Comment (Tobbe) [inline packages/project-config/package.json:21]: This one is `build.ts`, most others have been `build.mts`. What's the heuristics for picking the file extension?

File: packages/adapters/fastify/web/build.mts
Comment (Tobbe) [inline packages/adapters/fastify/web/build.mts:1]: I'm getting red squiggles on pretty much all build files
image

Can you please try to fix that?

**Review by Tobbe — CHANGES_REQUESTED** [review]
Can we please talk about how to fix the red squiggles in the build.mts files? I have an idea that I wanted to run by you

**Review by Tobbe — CHANGES_REQUESTED** [review]
Can we please talk about how to fix the red squiggles in the build.mts files? I have an idea that I wanted to run by you

File: packages/project-config/package.json (hunk 1 above)
Comment (jtoar) [reply packages/project-config/package.json:21]: This package is ESM so `.ts` files are ESM already here, they don't need the `m` prefix

File: packages/adapters/fastify/web/build.mts
Comment (orta) [reply packages/adapters/fastify/web/build.mts:1]: You want to have a higher `target` in typescript, given you have node 20 as the base minimum to ES2022 - https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping#node-20

Comment (jtoar) [reply packages/adapters/fastify/web/build.mts:1]: Thanks @orta! Just so you have full context, when @Tobbe and I paired on this we realized he had a TypeScript VSCode setting enabled I didn't:

image
 
I couldn't reproduce at the time but checking that box made the errors show up

Comment (orta) [reply packages/adapters/fastify/web/build.mts:1]: interesting!

**Pull request merged**